File: 001-010/003/003.cpp
#include <math.h>
#include <iostream>

// What is the largest prime factor of the number 600851475143?

int isPrime(int number) {
    for (int j = 2; j < number / 2; j++)
        if (number % j == 0) {
            return 0;
        }
    return 1;
}

int main() {
    double target = 600851475143;
    for (int i = 2; i <= target / 2; i++) {
        double k = target / i;
        if (ceilf(k) == k) {
            if (isPrime(static_cast<int>(k)) == 1) {
                std::cout << std::fixed << "The highest prime factorial is " << static_cast<int>(k) << "." << std::endl;
                break;
            }
        }
    }
}

File: 001-010/003/003.py
#!/usr/bin/env python

# What is the largest prime factor of the number 600851475143?

target = 600851475143


def isPrime(number):
    for j in range(2, int(number) // 2):
        if number % j == 0:
            return 1
    return 0


def main(target):
    for i in range(2, target // 2):
        k = target / i
        if k % 1 == 0:
            if isPrime(k) == 0:
                print("The highest prime factorial is", int(k))
                break


if __name__ == "__main__":
    main(target)
